### PR TITLE
Improve mypy

### PR DESCRIPTION
**Pull request by jonasteuwen**
Much stricter mypy passing. Eventually we want `--strict` passing, but I would like to merge these changes already to main.

### CODE REVIEW

File: dlup/tiling.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# coding=utf-8
```
Comment (EricMarcus-ai) [inline dlup/tiling.py:1]: Is there a particular reason for removing the codings in many files?

File: tests/test_image.py [hunk 2]
```diff
@@ -123,7 +122,8 @@ def test_read_region(
 
         # Check that the output corresponding shape and value.
         assert np.asarray(pil_extracted_region).shape == np.asarray(extracted_region).shape
-        assert np.allclose(pil_extracted_region, extracted_region)
+        print(np.asarray(pil_extracted_region).sum(), np.asarray(extracted_region).sum())
```
Comment (EricMarcus-ai) [inline tests/test_image.py:125]: Is this correct? Why remove the assert?

**Review by EricMarcus-ai — CHANGES_REQUESTED** [review]
Nice additions. Some small changes regarding imports from outside package

File: dlup/data/dataset.py [hunk 3]
```diff
@@ -258,15 +280,19 @@ def __getitem__(self, index):
...
         region = region_view.read_region(coordinates, region_size)
 
-        sample: StandardTilingFromSlideDatasetSample = {
+        sample: TileSample = {
             "image": region,
             "coordinates": coordinates,
             "mpp": mpp,
             "path": self.path,
             "region_index": region_index,
+            "labels": None,
```
Comment (EricMarcus-ai) [inline dlup/data/dataset.py:289]: Should we add these None variables by default? If we then don't have any labels or annotations in the dataset, we will end up with None in our batches, which torch will complain about when collating. Probably adding them only if self.annotations/labels is not None works more smoothly

File: dlup/tiling.py (hunk 1 above)
Comment (jonasteuwen) [reply dlup/tiling.py:1]: Yeah, it's not needed in python 3.

File: tests/test_image.py (hunk 2 above)
Comment (jonasteuwen) [reply tests/test_image.py:125]: Fixed

File: dlup/data/dataset.py (hunk 3 above)
Comment (jonasteuwen) [reply dlup/data/dataset.py:289]: Yes, otherwise you can’t make it a TypedDict

**Comment by jonasteuwen** [comment]
Addressed most of your comments @EricMarcus-ai

File: dlup/utils/mask.py [hunk 4]
```diff
@@ -53,7 +53,9 @@ def _DFS(
         sibling_id = hierarchy[sibling_id][0]
 
 
-def mask_to_polygons(mask: np.ndarray, offset: tuple[int, int] = (0, 0), scaling: float = 1.0) -> list[Polygon]:
+def mask_to_polygons(
```
[the remark this answers is not in the document]
Comment (jonasteuwen) [reply dlup/utils/mask.py:58]: It had to do with an opencv update.

File: dlup/background.py [hunk 5]
```diff
@@ -30,6 +30,7 @@
 from dlup import SlideImage
 from dlup._exceptions import DlupError
 from dlup.annotations import WsiAnnotations
+from utils.types import MaskTypes
```
[the remark this answers is not in the document]
Comment (jonasteuwen) [reply dlup/background.py]: Yeah, was a mistake

File: dlup/data/experimental/dataset.py [hunk 6]
```diff
@@ -7,12 +7,14 @@
...
 import numpy as np
+import numpy.typing as npt
 
 from dlup import SlideImage
 from dlup.annotations import WsiAnnotations
-from dlup.data.dataset import ROIType, TiledROIsSlideImageDataset, parse_rois
+from dlup.data.dataset import RegionFromSlideDatasetSample, ROIType, TiledROIsSlideImageDataset, parse_rois
 from dlup.experimental_backends import ImageBackend
 from dlup.tiling import Grid, GridOrder, TilingMode
+from utils.types import MaskTypes
```
[the remark this answers is not in the document]
Comment (jonasteuwen) [reply dlup/data/experimental/dataset.py]: Was a mistake

**Pull request merged**